internal/pkg/gateway/registry.go

```diff
@@ -312,18 +323,19 @@ func (reg *registry) registerChannel(channel string) error {
 	peers := map[string]string{}
 	members := reg.discovery.PeersOfChannel(common.ChannelID(channel))
 	for _, member := range members {
+		// TODO why is this fragile? Make it less fragile?
```

Comment:
Matt was never happy about this - but this code was copied from the discovery service

Reply:
I'll leave it as is for now then